pumpkin/src/command/commands/cmd_op.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{
+    command::{
+        args::{arg_players::PlayersArgumentConsumer, Arg, ConsumedArgs},
+        tree::CommandTree,
+        tree_builder::{argument, require},
+        CommandError, CommandExecutor, CommandSender,
+    },
+    entity::player::PermissionLvl,
+    server::json_config::{SaveJSONConfiguration, OPERATOR_CONFIG},
+};
+use async_trait::async_trait;
+use pumpkin_config::{op::Op, BASIC_CONFIG};
+use pumpkin_core::text::TextComponent;
+use CommandError::InvalidConsumption;
+
+const NAMES: [&str; 1] = ["op"];
+const DESCRIPTION: &str = "Specifies one or more game profiles (player profiles). Must be a player name (should be a real one if the server is in online mode) or a player-type target selector";
+const ARG_TARGET: &str = "player";
+
+struct OpExecutor;
+
+#[async_trait]
+impl CommandExecutor for OpExecutor {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), CommandError> {
+        let mut config = OPERATOR_CONFIG.write().await;
+
+        let Some(Arg::Players(targets)) = args.get(&ARG_TARGET) else {
+            return Err(InvalidConsumption(Some(ARG_TARGET.into())));
+        };
+
+        // log each player to the console.
+        for player in targets {
+            let op_entry = Op {
+                uuid: player.gameprofile.id,
+                name: player.gameprofile.name.clone(),
+                level: BASIC_CONFIG.op_permission_level,
```

Comment:
Maybe make it possible to set what level of op the player should be with cmd args.